From 533907539357df45d1923638d11192b98f7bf280 Mon Sep 17 00:00:00 2001
From: Darkvater <tamas.farago@credit-suisse.com>
Date: Sun, 4 May 2014 13:51:29 +0800
Subject: [PATCH] Update support for LG2012 TV

LG 2012 series does not need '.' in title anymore to be able to show
captions
---
 clients.c  | 8 ++++++++
 clients.h  | 1 +
 upnpsoap.c | 1 +
 3 files changed, 10 insertions(+)

diff --git a/clients.c b/clients.c
index 39ca362..cd1d06a 100644
--- a/clients.c
+++ b/clients.c
@@ -135,6 +135,14 @@ struct client_type_s client_types[] =
 	  EXAVClientInfo
 	},
 
+	/* USER-AGENT: Linux/2.6.35 UPnP/1.0 DLNADOC/1.50 INTEL_NMPR/2.0 LGE_DLNA_SDK/1.6.0 */
+	{ ELGNetCastDevice,
+	  FLAG_DLNA | FLAG_CAPTION_RES,
+	  "LG",
+	  "LGE_DLNA_SDK/1.6.0",
+	  EUserAgent
+	},
+
 	/* User-Agent: Linux/2.6.31-1.0 UPnP/1.0 DLNADOC/1.50 INTEL_NMPR/2.0 LGE_DLNA_SDK/1.5.0 */
 	{ ELGDevice,
 	  FLAG_DLNA | FLAG_CAPTION_RES,
diff --git a/clients.h b/clients.h
index 190d569..35d819a 100644
--- a/clients.h
+++ b/clients.h
@@ -57,6 +57,7 @@ enum client_types {
 	EDirecTV,
 	EFreeBox,
 	ELGDevice,
+	ELGNetCastDevice,
 	ELifeTab,
 	EMarantzDMP,
 	EMediaRoom,
diff --git a/upnpsoap.c b/upnpsoap.c
index 0f19035..0c2b9af 100644
--- a/upnpsoap.c
+++ b/upnpsoap.c
@@ -1055,6 +1055,7 @@ callback(void *args, int argc, char **argv, char **azColName)
 					break;
 				case ESamsungSeriesCDE:
 				case ELGDevice:
+				case ELGNetCastDevice:
 				case EAsusOPlay:
 				default:
 					if( passed_args->flags & FLAG_HAS_CAPTIONS )
-- 
2.11.0

